packages/react-server/core/ClientController.js

```diff
@@ -172,27 +243,6 @@ class ClientController extends EventEmitter {
 		context.onNavigate( (err, page, path, type) => {
 			logger.debug('Executing navigate action');
 
-			// if this is a History.events.PUSHSTATE navigation, we should change the URL in the bar location bar
-			// before rendering.
-			// note that for browsers that do not have pushState, this will result in a window.location change
-			// and full browser load. It's kind of late to do that, as we may have waited for handleRoute to
-			// finish asynchronously. perhaps we should have an "URLChanged" event that happens before "NavigateDone".
-			if (type === History.events.PUSHSTATE && this._history) {
-				// See "Sorry folks", above.
-				if (!(history.state||{}).reactServerFrame){
-					this._history.replaceState({
-						reactServerFrame: pushFrame(),
-					}, null, location.path);
-				}
-				this._history.pushState({
-					reactServerFrame: pushFrame(),
-				}, null, path);
-			} else if (type === History.events.PAGELOAD && this._history && this._history.canClientNavigate()) {
-				this._history.replaceState({
-					reactServerFrame: pushFrame(),
-				}, null, path);
-			}
-
```

Comment:
We were pushing state after running `handleRoute`.  Moved it up into `_startRequest`.